src/Streamly/Internal/Data/Producer.hs

```diff
@@ -0,0 +1,235 @@
+-- |
+-- Module      : Streamly.Internal.Data.Producer
+-- Copyright   : (c) 2021 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- A 'Producer' is an 'Unfold' with an 'extract' function added to to extract
+-- the state. It is more powerful but less general than an Unfold.
```

Comment:
Ah, so we agree that `Producer` is more powerful.

Reply:
Yes.